compiler/src/iree/compiler/Dialect/Flow/Transforms/test/dispatch_linalg_on_tensors_default.mlir

```diff
@@ -52,3 +52,35 @@ func.func @elem_set_encoding(%arg0: tensor<512xf32>, %arg1: tensor<384x512xf32>,
 // CHECK:           linalg.generic
 // CHECK:           iree_linalg_ext.set_encoding
 // CHECK-NOT:     flow.dispatch.workgroups
+
+// -----
+
+// Tests that we don't fuse transposed elementwise ops--it will require extra allocation from bufferization to hold the matmul result.
+
+func.func @transposed_elementwise(%4: tensor<4096x512xf16>, %5: tensor<512x512xf16>, %6: tensor<512xf16>) -> tensor<512x4096xf16> {
+  %f0 = arith.constant 0.000000e+00 : f16
+  %empty0 = tensor.empty() : tensor<512x4096xf16>
+  %empty1 = tensor.empty() : tensor<4096x512xf16>
+  %fill = linalg.fill ins(%f0 : f16) outs(%empty1 : tensor<4096x512xf16>) -> tensor<4096x512xf16>
+  %matmul = linalg.matmul
+    ins(%4, %5 : tensor<4096x512xf16>, tensor<512x512xf16>)
+    outs(%fill : tensor<4096x512xf16>) -> tensor<4096x512xf16>
+  %generic = linalg.generic {
```

Comment:
Actually I didnt pay attention to the lit test... This kind of fusion should be valid. The store is transposed, so this should be handlable in the backend... So this is a backend issue AFAICS.